### ISSUE TITLE
Error: exit status 1 when running generate on root

### ISSUE THREAD

**Issue by syphernl** [transcend-io/terragrunt-atlantis-config#220]
We've introduced this tool in to our toolchain when we needed to implement Terragrunt (and as a result, Atlantis). 

We are using the following command to generate the `atlantis.yml` which runs within the `pre_workflow_hooks`:

```bash
terragrunt-atlantis-config generate --create-workspace --autoplan --parallel --workflow terragrunt --terraform-version "$(head -n1 .terraform-version)" --output atlantis.yaml
```

Our directory structure looked like this:

```bash
├── _envcommon
├── modules
├── production
│   ├── global
└── staging
    ├── eu-west-1
    ├── global
    └── us-east-1
```

This worked fine as `terragrunt-atlantis-config` generated the configuration which was picked up by Atlantis. 

The `production` environment only contained `dns` and `dnssec` modules with dependencies to the `staging` env to allow it to add the required records for staging in the root domain like so:
```hcl
# To retrieve the nameserver records for the zone
dependency "dns_staging" {
  config_path = "${dirname(get_terragrunt_dir())}/../../staging/global/dns"
}

# To retrieve the DS record for the zone
dependency "dnssec_staging" {
  config_path = "${dirname(get_terragrunt_dir())}/../../staging/global/dnssec"
}
```

This worked fine, no issues whatsoever.

Today we copied all staging components to setup the actual production environment, making the directory structure look like this:

```bash
├── _envcommon
├── modules
├── production
│   ├── eu-west-1
│   ├── global
│   └── us-east-1
└── staging
    ├── eu-west-1
    ├── global
    └── us-east-1
```

Pretty straight-forward. 

If we run the command in the root of the project now it shows a lot of "Created project for...". 
However, at some point it suddenly exits without any explanation:

```
INFO[0013] Created project for /data/infra/staging/eu-west-1/cloudfront-assets/terragrunt.hcl 
INFO[0013] Created project for /data/infra/staging/eu-west-1/rds/terragrunt.hcl 
INFO[0013] Created project for /data/infra/staging/eu-west-1/cloudfront/terragrunt.hcl 
Error: exit status 1
```

Commenting out the dependencies from prod dns => staging dns has no effect on this.

If we run the command from either `staging` or `production` directory it generates an `atlantis.yaml` just fine (similarly to adding add  `--root staging` or `--root production` to the command). 

It does not error if we add `--cascade-dependencies=false` to the command, however this means we are missing out on some dependencies which should trigger a fresh plan/apply.

Is there something wrong with our structure or approach?

**Comment by dmattia** [comment]
That approach seems good overall. Would you be able to create a PR with a minimal example that has this error in this folder directory: https://github.com/transcend-io/terragrunt-atlantis-config/tree/master/test_examples? I can then create an integration test that attempts to build that folder and can update the code until it completes as expected

**Comment by syphernl** [comment]
It appears this problem was being caused by a workaround we have had in place for #196, but since this particular issue has been fixed in #202 the workaround can likely be removed which should solve this issue.

**Comment by dmattia** [comment]
Wonderful, I'll close this for now then 👍
